libvirt/resource_libvirt_dir_pool.go

```diff
@@ -0,0 +1,240 @@
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"capacity": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"allocation": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"available": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"xml": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"xslt": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceLibvirtDirPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+	if client.libvirt == nil {
+		return fmt.Errorf(LibVirtConIsNil)
+	}
+
+	poolName := d.Get("name").(string)
+
+	client.poolMutexKV.Lock(poolName)
+	defer client.poolMutexKV.Unlock(poolName)
+
+	// Check whether the storage pool already exists. Its name needs to be
+	// unique.
+	if _, err := client.libvirt.LookupStoragePoolByName(poolName); err == nil {
+		return fmt.Errorf("storage pool '%s' already exists", poolName)
+	}
+	log.Printf("[DEBUG] Pool with name '%s' does not exist yet", poolName)
+
+	poolPath := d.Get("path").(string)
+	poolDef := libvirtxml.StoragePool{
+		// We only support "dir" type at the moment
+		Type: "dir",
+		Name: poolName,
+		Target: &libvirtxml.StoragePoolTarget{
+			Path: poolPath,
+		},
+	}
+	data, err := xmlMarshallIndented(poolDef)
+	if err != nil {
+		return fmt.Errorf("Error serializing libvirt storage pool: %s", err)
+	}
+	log.Printf("[DEBUG] Generated XML for libvirt storage pool:\n%s", data)
+
+	data, err = transformResourceXML(data, d)
+	if err != nil {
+		return fmt.Errorf("Error applying XSLT stylesheet: %s", err)
+	}
+
+	// create the pool
+	pool, err := client.libvirt.StoragePoolDefineXML(data, 0)
+	if err != nil {
+		return fmt.Errorf("Error creating libvirt storage pool: %s", err)
+	}
+	defer pool.Free()
+
+	err = pool.Build(0)
+	if err != nil {
+		return fmt.Errorf("Error building libvirt storage pool: %s", err)
+	}
+
+	err = pool.SetAutostart(true)
+	if err != nil {
+		return fmt.Errorf("Error setting up libvirt storage pool: %s", err)
+	}
+
+	err = pool.Create(0)
+	if err != nil {
+		return fmt.Errorf("Error starting libvirt storage pool: %s", err)
+	}
+
+	err = pool.Refresh(0)
+	if err != nil {
+		return fmt.Errorf("Error refreshing libvirt storage pool: %s", err)
+	}
+
+	id, err := pool.GetUUIDString()
+	if err != nil {
+		return fmt.Errorf("Error retrieving libvirt pool id: %s", err)
```

Comment:
i was thinking it since we hardcode the `dir` type withing the implementation, if also the error-msg should be adapted

Reply:
with "Error retrieving libvirt directory pool id: %s", err)

Reply:
No, i don't think so. As I said before, quite a bit of this code can be moved to a common pool module, when someone adds another pool implementation.

Reply:
Also i don't think the exact type of the pool is relevant here.

Reply:
yop,  i think we can mark as resolved.